ansible-ipi-install/roles/installer/tasks/main.yml

```diff
@@ -45,6 +45,8 @@
   tags: customfs
 - include_tasks: 59_cleanup_bootstrap.yml
   tags: cleanup
+- include_tasks: 57_check_vip_status.yml
```

Comment:
Based on file name it looks like it should be executed before, should it be using 59_1_check_vip_status or something like that?

Reply:
actually we just want 57_ to be in front of 59_

Reply:
> > The logic behind
> > ```
> > - name: check if {{ apivip }} is accessible
> >   wait_for:
> >     host: '{{ apivip }}'
> >     port: 6443
> >     state: present
> >     delay: 0
> >     timeout: 10
> >   tags: vip_status
> > ```
> > 
> > 
> > seems to be more reasonable test compared to `ping` check.
> 
> @DirectedSoul1 `wait_for` would be used if you needed to wait for the API or Ingress VIP to come up but your not waiting for it to come up, what you need is to check if a particular URI is accessible or not.
> 
> Take a look at: https://docs.ansible.com/ansible/latest/modules/uri_module.html . This is what you need.

Yeah I was trying the same when I try it with `curl http://192.168.111.3:6443 -v` I'm getting the return code as `400` instead of `200` (control plane is up right now) which results in the module as shown below 
```
- name: Check the return status of {{ apivip }}
   uri:
     url: "http://{{ apivip }}:6443"
     return_content: yes
     status_code: 200  #default 200 
     register: result
  tags: validation

```

results in(maybe due to my disconnected environment) It would default to check for status code `200`  

```
{"changed": false, "content": "Client sent an HTTP request to an HTTPS server.\n", "elapsed": 0, "msg": "Status code was 400 and not [200]: HTTP Error 400: Bad Request", "redirected": false, "status": 400, "url": "http://192.168.111.3:6443"
```